### PR TITLE
Liudmylatkachenkooffcl dev

### PR DESCRIPTION
**Pull request by liudmylatkachenkooffcl**
zrobione zadanie 2. Do titanic_model.ipynb ładujemy titanic_clean_final.csv; Też zmodyfikowałam ubiegłe zadanie zgodnie z uwagami i sugestiami. Przy wątpliwościach piszczie mi na ds'a.

### CODE REVIEW

**Comment by liudmylatkachenkooffcl** [comment]
@Dominik-Galus jak coś to już w tym pull-u zrobiłam update 1 zadania. Info jest w titanic_info.md

**Review by Justa555 — COMMENTED** [review]
Nie wiem do końca jak mam zaliczać regresję logistyczną, bo myślałam, że wybieramy modele spośród tych podanych w poleceniu. Jeżeli regresja została uznana jako eksperyment, bo to rozumiem z jej opisu, to wtedy nie mamy drugiego modelu, a to było wymagane. Skupię się tylko na drzewie. Napewno z wymagań, których nie dostrzegłam to model nie jest odpalony na zbiorze treningowym i porównany z odpaleniem na zbiorze testowym. Brakuje badania hiperparametrów i wizualizacji tego przez wykres lub tabelkę. Nie rozumiem jeszcze komentarza w 100 komórce "#potem zwizualizuj to wszystko i pobadaj hiperparametrey bo idę na spacer.".  Dodatkowo nie jestem pewna, czy dobre jest usunięcie takiej ilości kolumn. Zdecydowanie na plus jest dodatkowe użycie walidacji krzyżowej.  Ogólnie cieżko mi ocenić zadanie ze względu na brak bądź użycie innego algorytmu niż wymagane.

**Review by jakub206 — COMMENTED** [review]
Ciekawy dobór modeli. Wybrałaś klasyczne drzewo oraz regresję logistyczną, która no nie była wymieniona w poleceniu więc za to plusik myślę. Mam sporą uwagę jednak dotyczącą badania hiperparametrów i jak zmiana ich wpływa na wybrane metryki (nie widzę wizualizacji oraz bezpośrednio gdzie to badasz)

**Review by higar4 — COMMENTED** [review]
Bardzo fajny pomysł z regresją logistyczną, tak naprawde nie widze czegos do czego mozna sie doczepic moze jedynie ze regresji logistycznej teoretycznie nie bylo do wyboru ale i tak fajny pomysl. Podoba mi sie jak to jest wszystko rozpisane fajnie sie to czyta

**Comment by Dominik-Galus** [comment]
Hej, poniżej podaje plusy i minusy

Plusy:
- Świadome użycie cross_val_score i random_state przy drzewie
- Wizualizacja drzewa i wag regresji logistycznej jest ciekawą wartością dodatkową
Minusy:
- nie widze zbadania hiperparametrów?
- brak standaryzacji dla regresji liniowej, fare i age mają różne skale a regresja logistyczna z L2 jest na to wrażliwa
- brak random_state w train_test_split, więc niereprodukowalny split jest
- deklarujesz f1 jako najważniejszą metryke ale wybierasz np. max_depth po accuracy a wnioski też dotyczą accuracy -> trzymaj się jednej lub pokaż dla obu

Ogólnie dużo rzeczy notatnik nie poruszył co było do wykonania w zadaniu
- dodaj gridsearchcv dla drzewa np. max_depth * min_samples_leaf dla regresji np. C * solver z wykresami zależności metryki od wartości
- dodaj random_state przy splicie
- zastosuj standardscaler dla regresji logistycznej
- przydałoby się ujednolicić narracje według jednej metryki albo pokazać dla wszystkich poruszanych
-  porównać train vs test accuracy, żeby ocenić przetrenowanie
